anyway/parsers/infographics_data_cache_updater.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+from ..utilities import init_flask
+from flask_sqlalchemy import SQLAlchemy
+from ..models import InfographicsDataCache, InfographicsDataCacheTemp, NewsFlash
+from ..constants import CONST
+import anyway.infographics_utils
+import logging
+
+
+app = init_flask()
+db = SQLAlchemy(app)
+
+
+def get_infographics_data_from_cache(news_flash_id, years_ago):
+    db_item = (
+        db.session.query(InfographicsDataCache)
+        .filter(InfographicsDataCache.news_flash_id == news_flash_id)
+        .filter(InfographicsDataCache.years_ago == years_ago)
+        .first()
+    )
+    logging.debug(f"retrieved from cache {type(db_item)}:{db_item}"[:70])
+    db.session.commit()
+    try:
+        return db_item.get_data() if db_item else {}
+    except Exception as e:
+        logging.error(
+            f"Exception while extracting data from returned cache item flash_id:{news_flash_id},years:{years_ago})"
+            f"returned value {type(db_item)}"
+            f":cause:{e.__cause__}, class:{e.__class__}"
+        )
+        return {}
+
+
+def copy_temp_into_cache():
+    num_items_cache = db.session.execute("SELECT * from infographics_data_cache").rowcount
+    num_items_temp = db.session.execute("SELECT * from infographics_data_cache_temp").rowcount
+    logging.debug(f"num items in cache: {num_items_cache}, temp:{num_items_temp}")
+    db.session.commit()
+    start = datetime.now()
+    with db.get_engine().begin() as conn:
+        conn.execute("lock table infographics_data_cache in exclusive mode")
+        logging.debug(f"in transaction, after lock")
+        conn.execute("delete from infographics_data_cache")
+        logging.debug(f"in transaction, after delete")
+        conn.execute(
+            "insert into infographics_data_cache SELECT * from infographics_data_cache_temp"
+        )
+        logging.debug(f"in transaction, after insert into")
+    logging.info(f"cache unavailable time: {str(datetime.now() - start)}")
+    num_items_cache = db.session.execute("SELECT * from infographics_data_cache").rowcount
```

Comment:
How much overhead do these queries to DB - for counts?
I see they are for logging, however if they have a large overhead, perhaps we can let them go? (I assume they are not a large overhead, but just want to make sure?)

Reply:
@ziv17 what about this comment?

Reply:
@ziv17 OK, I'm merging, I suggest you'll handle this one in a future PR